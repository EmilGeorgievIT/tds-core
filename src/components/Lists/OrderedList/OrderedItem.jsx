import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'

import textStyles from '../../Typography/Text/Text.modules.scss'
import styles from '../ListItem.modules.scss'

const OrderedItem = ({ size, children, ...rest }) => {
  const classes = `
    ${styles.item}
    ${textStyles[size]}
  `

  return (
    <li {...safeRest(rest)} className={classes}>
      {children}
    </li>
  )
}

OrderedItem.propTypes = {
  size: PropTypes.oneOf([
    'medium',
    'large'
  ]),
  children: PropTypes.node.isRequired
}

OrderedItem.defaultProps = {
  size: undefined
}

OrderedItem.displayName = 'OrderedList.Item'

export default OrderedItem
