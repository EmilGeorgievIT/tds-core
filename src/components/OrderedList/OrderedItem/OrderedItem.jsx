import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'

const OrderedItem = ({ children, ...rest }) => (
  <li {...safeRest(rest)}>
    {children}
  </li>
)

OrderedItem.propTypes = {
  /**
   * The content
   */
  children: PropTypes.node.isRequired
}

OrderedItem.displayName = 'OrderedList.Item'

export default OrderedItem
