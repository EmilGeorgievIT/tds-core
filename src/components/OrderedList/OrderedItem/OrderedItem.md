```
<OrderedList>
  <OrderedList.Item>As a Cisco Gold partner, get TELUS support 24/7 from fully trained Cisco agents.</OrderedList.Item>
  <OrderedList.Item>Protect your business with a world-class end-to-end solution, designed around an advanced foundation of security, privacy and reliability.</OrderedList.Item>
  <OrderedList.Item>Share desktops and collaborate in real time, quickly and conveniently.</OrderedList.Item>
</OrderedList>
```
