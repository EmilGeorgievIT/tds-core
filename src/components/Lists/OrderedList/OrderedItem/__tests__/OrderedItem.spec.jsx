import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import OrderedItem from '../OrderedItem'

describe('OrderedItem', () => {
  const defaultChildren = 'Some content'
  const doShallow = (props = {}, children = defaultChildren) => (
    shallow(<OrderedItem {...props}>{children}</OrderedItem>)
  )

  it('<OrderedItem />', () => {
    const item = doShallow()

    expect(toJson(item)).toMatchSnapshot()
  })

  it('renders an HTML li tag', () => {
    const item = doShallow()

    expect(item).toHaveTagName('li')
  })

  it('passes additional attributes to the list item', () => {
    const item = doShallow({ id: 'the-item' })

    expect(item).toHaveProp('id', 'the-item')
  })

  it('does not allow custom CSS', () => {
    const item = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(item).not.toHaveProp('className', 'my-custom-class')
    expect(item).not.toHaveProp('style')
  })
})
