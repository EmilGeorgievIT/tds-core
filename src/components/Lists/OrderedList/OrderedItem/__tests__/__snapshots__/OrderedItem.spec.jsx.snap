// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderedItem <OrderedItem /> 1`] = `
<li>
  Some content
</li>
`;
